lib/slack/real_time/config.rb

```diff
@@ -15,6 +18,23 @@ def reset
         self.websocket_ping = 30
         self.websocket_proxy = nil
         self.token = nil
+        self.concurrency = method(:detect_concurrency)
+      end
+
+      def concurrency
+        (val = @concurrency).respond_to?(:call) ? val.call : val
+      end
+
+      def detect_concurrency
+        [:Eventmachine, :Celluloid].each do |concurrency|
+          begin
+            return Slack::RealTime.const_get(concurrency)
+          rescue LoadError
+            # could not be loaded, missing dependencies
+          end
+        end
+
+        raise NoConcurrencyError, 'could not find concurrency primitives, add faye-websocket or celluloid-io'
```

Comment:
The meta dependency is a pretty neat idea. But it would require 2 releases every time and I am not sure it's such a great idea.

Also, btw, maybe if you have *both* you have to choose explicitly? There should probably be tests for that.